### PR TITLE
mat64: be strict with triangle for SymDense

### LINKED ISSUE #261: mat64: SymDense has paths coded that are not reachable
We don't support lower symmetric matrices via mat64, though we do via blas. However there are paths in Max, Min that cannot be reached because they examing the lower triangle of the data slice.

In itself this is not a problem, but that code cannot easily be tested (list_test tests won't allow it because they depend on CopySym which will panic when handed a lower-encoded symmetric matrix) and I don't see any real reason to keep it if there is no easy way to create a SymDense in this state.

When SetRawSymmetric is added it should panic if the blas64.Symmetric is lower (a similar check for blas.Unit should be done for SetRawTriangular when it appears).

### PR DESCRIPTION
**Pull request by kortschak**
Fixes #261.

@btracey Please take a look.

### CODE REVIEW

File: mat64/symmetric.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ var (
 
 const badSymTriangle = "mat64: blas64.Symmetric not upper"
 
-// SymDense is a symmetric matrix that uses Dense storage.
+// SymDense is a symmetric matrix that uses dense storage. SymDense
+// matrices are encoded in upper the upper triangle.
```
Comment (btracey) [inline mat64/symmetric.go]: in upper triangular format?

Comment (btracey) [inline mat64/symmetric.go]: or just 'in the upper triangle'. The current sentence isn't english.

**Comment by btracey** [comment]
LGTM with comment fixed.

**Comment by kortschak** [comment]
Multitasking is a bad thing.

File: mat64/symmetric.go (hunk 1 above)
Comment (kortschak) [inline mat64/symmetric.go]: Done.

**Pull request merged**